frontend/src/components/PrefillTemplateColumns.tsx

```diff
@@ -0,0 +1,203 @@
+            delete formErrors_copy[key];
+            setFormErrors({ ...formErrors_copy })
+        }
+    }, [formErrors])
+
+    const onValuesChange: NonNullable<FormProps['onValuesChange']> = useCallback((changedValues: any) => {
+        const key = Object.keys(changedValues)[0]
+        checkFormErrors(key)
+    }, [checkFormErrors])
+
+    const returnPrefillData = useCallback(() => {
+        const fieldValues = form.getFieldsValue();
+        const prefillData: Record<string, string> = {}
+        const errorData = {}
+        let error = false
+        Object.keys(fieldValues).forEach((column) => {
+            if (checkedFields[column]) {
+                console.debug(column, fieldValues[column])
+                if (fieldValues[column] == undefined) {
+                    errorData[column] = 'Missing Field'
+                    error = true
+                } else {
+                    if (!column.toLocaleLowerCase().includes("date")) {
+                        prefillData[column] = fieldValues[column]
+                    } else {
+                        prefillData[column] = new Date(fieldValues[column]).toISOString().split('T')[0]
+                    }
+                }
+            }
+        })
+        if (error) {
+            setFormErrors(errorData)
+            return null
+        }
+        return prefillData
+    }, [checkedFields, form])
+
+    const onFinish: NonNullable<FormProps['onFinish']> = useCallback(() => {
+        const prefillData = returnPrefillData()
+        if (prefillData) {
+            handlePrefillTemplate(prefillData)
+            setIsPrefillColumnsShown(false)
+        }
+    }, [handlePrefillTemplate, returnPrefillData])
+
+    useEffect(() => {
+        const fields = {}
+        Object.keys(data).forEach((column) => {
+            fields[column] = true;
+        })
+        setCheckedFields(fields)
+    }, [data])
+
+    const ColumnTypeHandlersCallback: (field: string) => ColumnTypeHandlers = useCallback((field: string) => ({
+        input: (type) => <Input type={type} disabled={!checkedFields[field]} style={{ textAlign: 'right' }} />,
+        date: () => <DatePicker disabled={!checkedFields[field]} />,
+        qcInstrument: () => <SelectInstrumentType type={'Quality Control'} disabled={!checkedFields[field]} style={{ textAlign: 'left' }} />,
+        select: (options) => <Select options={options.map((o) =>  ({ label: o, value: o }))} disabled={!checkedFields[field]} style={{ textAlign: 'left' }}/>
+    }), [checkedFields])
+
+    return (
+        <>
+            <Button type='primary' disabled={!canPrefill} onClick={Object.keys(data).length === 0 ? () => handlePrefillTemplate({}) : showPrefillColumns} title='Download a prefilled template with the selected samples'>Prefill Template</Button>
+            <Modal title={"Prefilled Columns"} visible={isPrefillColumnsShown} okText={"Prefill"} onOk={form.submit} onCancel={cancelPrefillTemplate}>
```

Comment:
The modal window is smallish and the value chosen from the drop down do not appear entirely.
There is too much space for the checkbox (to its right) compared to the rest of the line content (which is the most important part).